packages/layout/src/layout.utils.ts

```diff
@@ -0,0 +1,274 @@
+import { get as getFromTheme } from "@styled-system/css"
+import { isArray } from "@chakra-ui/utils"
+
+const __DEV__ = process.env.NODE_ENV !== "production"
+
+/**
+ * convertToArray takes a given style property, like `lineHeight`, and converts it into an
+ * equivalent styled-system responsive array
+ *
+ * @param theme - the styled-system theme object
+ * @param value - the style value to convert to an array
+ *
+ * @example Examples can be found in the associated test file
+ */
+export function convertToArray<T>(
+  theme: object,
+  value: T | T[] | { [key: string]: T },
+): T[] {
+  const breakpoints: string[] & { [key: string]: string } = getFromTheme(
+    theme,
+    "breakpoints",
+  )
+
+  if (isArray(value)) {
+    return value
+  } else if (value === undefined) {
+    return []
+  } else if (typeof value === "object") {
+    // In order for the text crop feature to work, we must know the _order_ of the breakpoints,
+    // meaning we require the breakpoints to be an array (rather than an object)
+    // Note: aliases as members of the array are still supported
+    if (!isArray(breakpoints)) {
+      throw new Error(
```

Comment:
Amazing! Great work @Vynlar.

One thing: we're actually looking to enforce object breakpoints going forward (for better intuitiveness). Most users find it difficult to understand the aliasing part.

I see you're internally converting the object to an array, so I thought we can avoid throwing.

Reply:
Are you saying that you plan to require users define breakpoints like this?:
```
breakpoints: {
sm: "30em",
md: "48em",
lg: "62em",
xl: "80em"
}
```
Or as an array?:
```
breakpoints: ["30em", "48em", "62em", "80em"]
```

Reply:
Yes. Like this:
```js
breakpoints: {
sm: "30em",
md: "48em",
lg: "62em",
xl: "80em"
}
```

Reply:
Thanks for the clarification.

In order for me to calculate these text crop values, I need to know the **order** of the breakpoints. If they are defined as an object, I no longer have a way to determine the order. Perhaps we could create a helper function for users to use in their theme files?:
```
breakpoints: makeBreakpoints([
  ["sm", "30em"],
  ["md", "48em"],
  // and so on...
])
```